src/uwtools/config/validator.py

```diff
@@ -1,22 +1,28 @@
 """
 Support for validating a config using JSON Schema.
 """
+
 import json
 import logging
 from pathlib import Path
 from typing import List
 
 import jsonschema
 
-from uwtools.config import YAMLConfig
+from uwtools.config.core import YAMLConfig
+from uwtools.types import DefinitePath, OptionalPath
 
 # Public functions
 
 
-def config_is_valid(config_file: str, schema_file: str) -> bool:
+def validate_yaml(schema_file: DefinitePath, config_file: OptionalPath = None) -> bool:
```

Comment:
I'm waffling on whether to write a ticket to extend this functionality to all our config file types. We're quite a way away from wanting to validate NML files, and I am not so sure I want to be on the hook for validating INI files at this point either.

Reply:
I'm not really sure it makes sense to try to validate anything but our own DSL YAML. I mean, what would it mean to validate a namelist file? To do so, we'd have to know what namelist names and variable names are accepted by the component, and what valid values look like. It's possible that we can validate Rocoto XML files because we have a schema for those. Where's the schema for an FV3 namelist file? And what if a scientist wants to add a Fortran variable for some tests they're doing and expose that variable in a namelist file? Their (perfectly valid w.r.t. their test Fortran code) namelist file would be rejected by `uwtools`, unless they also updated its schema, which prospect I do not find realistic. I'm not sure it's really our business to validate the configuration files of software we don't control (except maybe Rocoto, since it should be nearly trivial to do, and since it's unlikely that anyone using `uwtools` would be changing it).